File: frontend/src/app/models/tipodocumento.ts
export class Tipodocumento {
  id_tipo_documento?: number;
  codigo: string;
  nombre: string;
  descripcion: string;
  estado: boolean;

  constructor(id_tipo_documento: number, codigo: string, nombre: string, descripcion: string) {
    this.id_tipo_documento = id_tipo_documento;
    this.codigo = codigo;
    this.nombre = nombre;
    this.descripcion = descripcion;
  }
}

File: backend/src/main/java/com/example/backend/model/Tipodocumento.java
package com.example.backend.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import lombok.Data;

@Data
@Entity
public class Tipodocumento {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id_tipo_documento;

	@Column(name = "codigo", nullable = false, unique = true)
	private String codigo;

	@Column(name = "nombre", nullable = true, unique = true)
	private String nombre;

	@Column(name = "descripcion", nullable = false)
	private String descripcion;

	@Column(name = "estado", nullable = true)
	private Boolean estado;

	

}
